Sources/KeyboardSupport/KeyboardScrollable.swift

```diff
@@ -0,0 +1,210 @@
+        if let keyboardWillShowObserver = keyboardWillShowObserver {
+            NotificationCenter.default.removeObserver(keyboardWillShowObserver)
+        }
+        if let keyboardWillHideObserver = keyboardWillHideObserver {
+            NotificationCenter.default.removeObserver(keyboardWillHideObserver)
+        }
+    }
+    
+    func keyboardWillShow(keyboardInfo: KeyboardInfo) {
+        // No-op by default. Opt-in by implementing this method in your class conforming to KeyboardScrollable.
+    }
+    
+    func keyboardWillHide(keyboardInfo: KeyboardInfo) {
+        // No-op by default. Opt-in by implementing this method in your class conforming to KeyboardScrollable.
+    }
+    
+    // MARK: Private Methods
+    
+    private func adjustViewForKeyboardAppearance(with keyboardInfo: KeyboardInfo, firstResponder: UIView) {
+        guard let scrollView = keyboardScrollableScrollView else { return }
+        
+        var mutableInset: UIEdgeInsets
+        if preservesContentInsetWhenKeyboardVisible, let originalContentInset = scrollView.originalContentInset {
+            mutableInset = originalContentInset
+        } else {
+            mutableInset = .zero
+        }
+        
+        // Adjust scroll view insets for keyboard height
+        let keyboardHeight = keyboardInfo.finalFrame.height
+        if #available(iOS 11.0, *) {
+            mutableInset.bottom += keyboardHeight - view.safeAreaInsets.bottom
+        } else {
+            mutableInset.bottom += keyboardHeight
+        }
+        
+        adjustScrollViewInset(mutableInset, keyboardInfo: keyboardInfo)
+        
+        // If active text field is hidden by keyboard, scroll so it's visible
+        if let textView = firstResponder as? UITextView {
+            scrollToSelectedText(for: textView, keyboardInfo: keyboardInfo)
+        } else {
+            
+            let preferredPaddingAroundInput = (firstResponder as? KeyboardPaddingProviding)?.inputPadding ?? .zero
+            scrollToRectIfNecessary(rect: firstResponder.frame, of: firstResponder, keyboardInfo: keyboardInfo, preferredPaddingAroundInput: preferredPaddingAroundInput)
+        }
+    }
+    
+    private func scrollToSelectedText(for textView: UITextView, keyboardInfo: KeyboardInfo) {
+        // Get the frame of the cursor/selection to improve scrolling position for UITextView's
+        // DispatchQueue.async() is necessary because the selectedTextRange typically hasn't not been updated when UIResponder.keyboardWillShowNotification is posted
+        DispatchQueue.main.async {
+            guard let textRange = textView.selectedTextRange, let selectionRect = textView.selectionRects(for: textRange).first else { return }
+            // Set an arbitray width to the target CGRect in case the width is zero. Otherwise, scrollRectToVisible has no effect.
+            self.scrollToRectIfNecessary(rect: selectionRect.rect.modifying(width: 30), of: textView, keyboardInfo: keyboardInfo)
```

Comment:
Is there a reason we picked 30 here instead of 1?

Reply:
I have not idea. It was something @br-johndavis added.

Reply:
Haha, I've noticed that multiple times and it's always bugged me. `30` seems too...arbitrary...lol. I feel like it should just be `1` if it needs to be non-zero.

Reply:
Either that or we should put like **_ARBITRARY_** :D

Reply:
It shall be 1